Porting regression tests to Orion

MinsukJi-NOAA opened the pull request:
Issue https://github.com/ufs-community/ufs-s2s-model/issues/77

- This branch has merged in the latest develop
- I will run 3 separate regression tests and add the result files when completed:
    1. NEMSCompsetRun on Hera
    2. rt.sh on Hera
    3. rt.sh on Orion

DeniseWorthen committed:
Merge pull request #1 from ufs-community/develop

Feature/add comparisons (#4)

DeniseWorthen committed:
Update .gitmodules

DeniseWorthen committed:
update submodules for nems,stochastic_physics

DeniseWorthen committed:
update NEMS,stochastic_physics

DeniseWorthen committed:
Merge pull request #2 from ufs-community/develop

merge develop/ufs-s2s-model

DeniseWorthen committed:
update CICE submode for feature/halfdeg (with build change) and
MOM_input template for initial condition files

DeniseWorthen committed:
Merge pull request #3 from ufs-community/develop

merge fv3atm/develop

DeniseWorthen committed:
Revert "update CICE submode for feature/halfdeg (with build change) and"

This reverts commit 0c8cb0b8451a11f7ab897cb4d31fc6233cf96c31.

DeniseWorthen committed:
update CICE submodule

DeniseWorthen committed:
update develop branch in my fork to use NOAA-EMC develop branches

DeniseWorthen committed:
Merge pull request #5 from ufs-community/develop

merge to latest develop

DeniseWorthen committed:
Merge pull request #6 from ufs-community/develop

Debug mode capablility (#15)

DeniseWorthen committed:
Update configure.fv3_coupled.hera.intel

DeniseWorthen committed:
Update configure.fv3_coupled.cheyenne.intel

DeniseWorthen committed:
Merge pull request #7 from ufs-community/develop

Add debug compilation flag specification at top level (#17)

DeniseWorthen committed:
Merge pull request #8 from ufs-community/develop

update to current ufs-weather-app (#19)

DeniseWorthen committed:
update gitmodules, FV3 for gfsv16 incomming commit, NEMS for
nomerg2atm, new MOM6 dev/emc

DeniseWorthen committed:
Revert changes that were meant for a branch, not develop

This reverts commit 57236a665d1efcbac28a53826daaa7a0385746eb.

DeniseWorthen committed:
Merge pull request #9 from ufs-community/develop

Update FV3, NEMS (#21)

DeniseWorthen committed:
Merge pull request #13 from ufs-community/develop

Update2020 01 (#23)

DeniseWorthen committed:
Merge pull request #15 from ufs-community/develop

DeniseWorthen committed:
Merge pull request #17 from ufs-community/develop

Update NEMS,FV3,CICE (#28)

DeniseWorthen committed:
Merge pull request #18 from ufs-community/develop

DeniseWorthen committed:
Merge pull request #19 from ufs-community/develop

Add CCPP to UFS-S2S-model

MinsukJi-NOAA committed:
first commit: tests/rt.sh works for cpld_fv3_mom6_cice_cold_atm_flux

DeniseWorthen committed:
Merge pull request #20 from ufs-community/develop

MinsukJi-NOAA committed:
Merge remote-tracking branch 'upstream/develop' into feature/alt_compsetrun

MinsukJi-NOAA committed:
Update submodules to match latest s2s develop's

DeniseWorthen committed:
Inital port to orion commit

MinsukJi-NOAA committed:
Add CPLWAV2ATM to default_vars. Change baseline to develop-20200326

DeniseWorthen committed:
add orion conf and modulefiles

DeniseWorthen committed:
update fv3_coupled for orion

MinsukJi-NOAA committed:
Add test case: cpld_fv3_mom6_cice_2d_atm_flux

DeniseWorthen committed:
update NEMS and FV3 submodules

MinsukJi-NOAA committed:
Add case cpld_fv3_ccpp_mom6_cice_cold_atm_flux. Modify rt.sh for create_baseline

MinsukJi-NOAA committed:
Modify rt.sh and rt_utils.sh to make 'create baseline' work

DeniseWorthen committed:
update components for orion build; update gitmodules
to point to relevant NOAA-EMC repos

DeniseWorthen committed:
update MOM6 again

DeniseWorthen committed:
Merge pull request #21 from MinsukJi-NOAA/feature/alt_compsetrun

DeniseWorthen committed:
update fv3_coupled

MinsukJi-NOAA committed:
Added all CCPP PROD tests, Orion changes

MinsukJi-NOAA committed:
Complete CCPP PRO unit tests on Hera
*Consolidated fv3_conf/*_run.IN files
*Modified tests/default_vars.sh and tests/TEST_NAME files
*RegressionTests_hera shows passing of all tests

MinsukJi-NOAA committed:
Add Orion-related config & module file

DeniseWorthen committed:
Merge pull request #22 from ufs-community/develop

Add debug (#70)

DeniseWorthen committed:
Merge branch 'develop' into feature/orion

DeniseWorthen committed:
update FV3

MinsukJi-NOAA committed:
Make Rocoto workflow manager work for S2S model

MinsukJi-NOAA committed:
Make ecflow workflow manager work for S2S model

DeniseWorthen committed:
update FV3,NEMS

MinsukJi-NOAA committed:
Modifications for Orion

MinsukJi-NOAA committed:
Merge remote-tracking branch 'mji/feature/alt_compsetrun' into feature/orion

MinsukJi-NOAA committed:
Complete RT porting to Orion

MinsukJi-NOAA committed:
Merge in latest develop branch

MinsukJi-NOAA committed:
Remove FV3 orion branch in .gitmodules

MinsukJi-NOAA commented:
RT passed on Orion, but all tests failed on Hera

junwang-noaa commented:
@MinsukJi-NOAA, I see your branch has the latest fv3 with physics updates,
these updates will change results. So we need to create a new baseline on
hera and Orion for this commit. @DeniseWorthen,  would you please create
PRs for mom6 and cice orion porting? Also since most of the near future
commits would be from CMPES, do you want to have the baselines temporarily
saved in your directory or it's OK in Minsuk's directory?


On Fri, May 1, 2020 at 6:53 PM Minsuk Ji <notifications@github.com> wrote:

> RT passed on Orion, but all tests failed on Hera
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-s2s-model/pull/78#issuecomment-622600804>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TONKI7CV5YUYJELUIDRPNHH7ANCNFSM4MXKY5RA>
> .
>

DeniseWorthen commented:
Both MOM6 and CICE5 have the porting updates made to the develop branches already. I will make PRs for ufs-s2s-model to update to those @develop. 

For the CMEPS commits, I need to better understand the order for the upcoming commits. Is wav->atm coupling coming first or is cmeps?

MinsukJi-NOAA commented:
I will close this PR, and open a new one when ufs-s2s-model develop branch is updated with latest components.

JessicaMeixner-NOAA commented:
> Both MOM6 and CICE5 have the porting updates made to the develop branches already. I will make PRs for ufs-s2s-model to update to those @develop.
> 
> For the CMEPS commits, I need to better understand the order for the upcoming commits. Is wav->atm coupling coming first or is cmeps?

This is unclear to me as well.  It's also unclear which regression test system and platforms will need to be updated and supported, etc.  Also I'm assuming an update for wcoss-phase2 might be coming soon as well?

DeniseWorthen commented:
I'm now really confused---I made a new PR for MOM6 and CICE5 to be updated with the orion ports but w/o FV3 updated. The last commit on FV3 now for the PR (#80) to ufs-s2s is the removal of the CCPP dynamic build. I thought this would allow you to use the existing hera baselines since I had updated my original port to orion ufs-s2s-model branch to the FV3 with the Chsp physics update. 

My original feature/orion port had updated FV3 because of an earlier email from Jun asking me to update my branch and I think I must have misunderstood which components I was being asked to update.

junwang-noaa commented:
I think at last Tuesday's meeting, the plan is to get Orion porting in
first, then we have a ocean-wave coupling commit (is it tested and ready
for commit yet?, it will be running on hera and orion. It can be run on
phase2 when phase2 porting is finished). Then we will have several commits
for CMEPS transition. For Orion porting, @ DeniseWorthen Minsuk' branch
updates all the subcomponents, since current FV3 and NEMS have CEMPS
related updates, do we need extra commits on CICE/MOM6 for s2s to run with
latest FV3/NEMS? If yes, then let's update s2s to point to the latest
version of all the components. If not, then let's keep FV3 at  aac9ff9
<https://github.com/NOAA-EMC/fv3atm/tree/aac9ff9da6914792f018f78c21d769c6336b8920>
(removal
ccpp dynamic build) and NEMS at  6e36027
<https://github.com/NOAA-EMC/NEMS/tree/6e36027d643e5153fc42694263bc7826dce2df08>
with
debug test changes, we will only make porting changes (updating mom6,
cice5 + s2s) in this commit.

On Mon, May 4, 2020 at 9:40 AM Denise Worthen <notifications@github.com>
wrote:

> I'm now really confused---I made a new PR for MOM6 and CICE5 to be updated
> with the orion ports but w/o FV3 updated. The last commit on FV3 now for
> the PR (#80 <https://github.com/ufs-community/ufs-s2s-model/pull/80>) to
> ufs-s2s is the removal of the CCPP dynamic build. I thought this would
> allow you to use the existing hera baselines since I had updated my
> original port to orion ufs-s2s-model branch to the FV3 with the Chsp
> physics update.
>
> My original feature/orion port had updated FV3 because of an earlier email
> from Jun asking me to update my branch and I think I must have
> misunderstood which components I was being asked to update.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-s2s-model/pull/78#issuecomment-623470325>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TMYV5SRKD6X3MX4NADRP3ATRANCNFSM4MXKY5RA>
> .
>

junwang-noaa commented:
Sorry, I confused myself, need a cup of coffee.  For Orion porting, @
DeniseWorthen Minsuk' branch updates all the subcomponents, since current
FV3 and NEMS have CEMPS related updates, do we need extra commits on
CICE/MOM6 for s2s to run with latest FV3/NEMS? If *NO*, then let's update
s2s to point to the latest version of all the components. If *YES*, then
let's keep FV3 at  aac9ff9
<https://github.com/NOAA-EMC/fv3atm/tree/aac9ff9da6914792f018f78c21d769c6336b8920>
(removal
ccpp dynamic build) and NEMS at  6e36027
<https://github.com/NOAA-EMC/NEMS/tree/6e36027d643e5153fc42694263bc7826dce2df08>
with
debug test changes, we will only make porting changes (updating mom6,
cice5 + s2s) in this commit. Hope it is clear now.


On Mon, May 4, 2020 at 9:57 AM Jun Wang - NOAA Federal <jun.wang@noaa.gov>
wrote:

> I think at last Tuesday's meeting, the plan is to get Orion porting in
> first, then we have a ocean-wave coupling commit (is it tested and ready
> for commit yet?, it will be running on hera and orion. It can be run on
> phase2 when phase2 porting is finished). Then we will have several commits
> for CMEPS transition. For Orion porting, @ DeniseWorthen Minsuk' branch
> updates all the subcomponents, since current FV3 and NEMS have CEMPS
> related updates, do we need extra commits on CICE/MOM6 for s2s to run with
> latest FV3/NEMS? If yes, then let's update s2s to point to the latest
> version of all the components. If not, then let's keep FV3 at  aac9ff9
> <https://github.com/NOAA-EMC/fv3atm/tree/aac9ff9da6914792f018f78c21d769c6336b8920> (removal
> ccpp dynamic build) and NEMS at  6e36027
> <https://github.com/NOAA-EMC/NEMS/tree/6e36027d643e5153fc42694263bc7826dce2df08> with
> debug test changes, we will only make porting changes (updating mom6,
> cice5 + s2s) in this commit.
>
> On Mon, May 4, 2020 at 9:40 AM Denise Worthen <notifications@github.com>
> wrote:
>
>> I'm now really confused---I made a new PR for MOM6 and CICE5 to be
>> updated with the orion ports but w/o FV3 updated. The last commit on FV3
>> now for the PR (#80
>> <https://github.com/ufs-community/ufs-s2s-model/pull/80>) to ufs-s2s is
>> the removal of the CCPP dynamic build. I thought this would allow you to
>> use the existing hera baselines since I had updated my original port to
>> orion ufs-s2s-model branch to the FV3 with the Chsp physics update.
>>
>> My original feature/orion port had updated FV3 because of an earlier
>> email from Jun asking me to update my branch and I think I must have
>> misunderstood which components I was being asked to update.
>>
>> —
>> You are receiving this because your review was requested.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/ufs-community/ufs-s2s-model/pull/78#issuecomment-623470325>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AI7D6TMYV5SRKD6X3MX4NADRP3ATRANCNFSM4MXKY5RA>
>> .
>>
>

DeniseWorthen commented:
MOM6 and CICE5 do not need to be updated to run with the latest FV3/NEMS (the cmeps related changes). 

The original feature/orion branch of s2s-model [https://github.com/DeniseWorthen/ufs-s2s-model/tree/feature/orion](https://github.com/DeniseWorthen/ufs-s2s-model/tree/feature/orion) has MOM6,CICE5 updated to develop in each case, which includes the orion ports. It also has FV3/NEMS updated to the current develop in each case (the cmeps related changes).

For CMEPS, I need to commit the restart features to CICE (1 commit to ufs-s2s, new baseline required). It can happen prior to the actual CMEPS commit; I have only been waiting until after the porting commits, the rt.sh commit and the dropping of ipd. 

To bring in CMEPS, a second commit would then be required to ufs-s2s (new baseline required) with the CMEPS related tests using rt.sh.

junwang-noaa commented:
Thanks for the clarification. So here is what we have: the latest revisions
of all the subcomponents have porting changes already, s2s can run when
pointing to the latest revisions of the subcomponent and RT results will
change because of the FV3 physics updates and ww3 code changes in develop
branch. For this Orion porting commit, we need to point to a ww3 revision
with porting updates, and it will change RT results anyway. So I would
suggest to use Minsuk's feature/orion branch (it already points to the
latest revisions of all the subcomponents), create new baseline on hera and
Orion and run rt.sh for RT and then commit the code. Minsuk, can you reopen
your PR?  Thanks

On Mon, May 4, 2020 at 10:17 AM Denise Worthen <notifications@github.com>
wrote:

> MOM6 and CICE5 do not need to be updated to run with the latest FV3/NEMS
> (the cmeps related changes).
>
> The original feature/orion branch of s2s-model
> https://github.com/DeniseWorthen/ufs-s2s-model/tree/feature/orion has
> MOM6,CICE5 updated to develop in each case, which includes the orion ports.
> It also has FV3/NEMS updated to the current develop in each case (the cmeps
> related changes).
>
> For CMEPS, I need to commit the restart features to CICE (1 commit to
> ufs-s2s, new baseline required). It can happen prior to the actual CMEPS
> commit; I have only been waiting until after the porting commits, the rt.sh
> commit and the dropping of ipd.
>
> To bring in CMEPS, a second commit would then be required to ufs-s2s (new
> baseline required) with the CMEPS related tests using rt.sh.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-s2s-model/pull/78#issuecomment-623491132>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TJ7AKVY4A2MUZQ5GBLRP3E6BANCNFSM4MXKY5RA>
> .
>

junwang-noaa commented on .gitmodules:
```diff
@@ -1,15 +1,12 @@
 [submodule "FV3"]
 	path = FV3
 	url = https://github.com/NOAA-EMC/fv3atm
-	branch = develop
```
please do not remove this "develop" branch. Some repo the default branch is "master", removing this line may cause problem.  This applied to line 8,  12, 26.

MinsukJi-NOAA replied:
FV3 branch was specified as "orion", which does not exist any more.
I will add "develop" to FV3, NEMS, CIC_SRC, MOM6, and WW3.

MinsukJi-NOAA committed:
Add develop to FV3,NEMS,CICE,MOM6,WW3 in gitmodules

MinsukJi-NOAA committed:
Add batch queue for ecflow on orion

MinsukJi-NOAA committed:
Add orion RT results

MinsukJi-NOAA committed:
Add rt.sh and nemscompsetrun results

MinsukJi-NOAA commented:
On Hera, if we generate Baseline using rt.sh, and run NEMSCompsetRun RT against it, there will be a file missing error for REGTEST-FINGERPRINT.md because rt.sh does not create this file when generating Baseline. Either we need to remove verify_fingerprint in NEMS/tests/rtgen, or stick to generating Baseline using NEMSCompsetRun.

DeniseWorthen reviewed (approved):
Looks good.

junwang-noaa commented:
Let's remove that. It tried to track the code changes with baseline. But it
affects model portability (I got problems on phase2 when there is no
matching baseline on the platform).

On Tue, May 5, 2020 at 7:07 AM Minsuk Ji <notifications@github.com> wrote:

> On Hera, if we generate Baseline using rt.sh, and run NEMSCompsetRun RT
> against it, there will be a file missing error for REGTEST-FINGERPRINT.md
> because rt.sh does not create this file when generating Baseline. Either we
> need to remove verify_fingerprint in NEMS/tests/rtgen, or stick to
> generating Baseline using NEMSCompsetRun.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-s2s-model/pull/78#issuecomment-623991063>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TPR25U272XXTIWRX3LRP7XO7ANCNFSM4MXKY5RA>
> .
>

MinsukJi-NOAA commented:
I will open an issue in NEMS repo regarding this.

junwang-noaa commented on compsets/all.input:
```diff
@@ -23,8 +23,8 @@ run cpld_fv3_ccpp_384_mom6_cice_1d_bmark_rt @ fv3, coupledapp, warm384, physics,
 
-run cpld_fv3_ccpp_mom6_cice_cold_debug @ fv3, coupledapp, cold, physics, debug
-run cpld_fv3_ccpp_mom6_cice_6h_debug @ fv3, coupledapp, warm, physics, debug
+#run cpld_fv3_ccpp_mom6_cice_cold_debug @ fv3, coupledapp, cold, physics, debug
```
May I ask why we comment out these two debug tests?

MinsukJi-NOAA replied:
Debug tests are not implemented in rt.sh yet. Since the new baselines (20200504) were generated using rt.sh, I had to comment those lines.

junwang-noaa commented:
I see, let's make a separate commit for the debug test then. At this time
we will keep running NEMSCompsetRun and rt.sh until the debug tests are in
rt.sh.

On Tue, May 5, 2020 at 8:46 AM Minsuk Ji <notifications@github.com> wrote:

> *@MinsukJi-NOAA* commented on this pull request.
> ------------------------------
>
> In compsets/all.input
> <https://github.com/ufs-community/ufs-s2s-model/pull/78#discussion_r420080248>
> :
>
> > @@ -23,8 +23,8 @@ run cpld_fv3_ccpp_384_mom6_cice_1d_bmark_rt @ fv3, coupledapp, warm384, physics,
>  run cpld_fv3_ccpp_384_mom6_cice_ww3_cold_bmark_rt @ fv3, coupledapp, coldBM, physics, bm, ww3
>  run cpld_fv3_ccpp_384_mom6_cice_ww3_1d_bmark_rt @ fv3, coupledapp, warm384, physics, bm, ww3
>
> -run cpld_fv3_ccpp_mom6_cice_cold_debug @ fv3, coupledapp, cold, physics, debug
> -run cpld_fv3_ccpp_mom6_cice_6h_debug @ fv3, coupledapp, warm, physics, debug
> +#run cpld_fv3_ccpp_mom6_cice_cold_debug @ fv3, coupledapp, cold, physics, debug
>
> Debug tests are not implemented in rt.sh yet. Since the new baselines
> (20200504) were generated using rt.sh, I had to comment those lines.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-s2s-model/pull/78#discussion_r420080248>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TK4PYGCZO3JHG667FLRQADBPANCNFSM4MXKY5RA>
> .
>

MinsukJi-NOAA commented:
For this PR, would it be safer if we generated the new baselines on Hera using nemscompsetrun? And start using rt.sh for baselines when the NEMS issue (verify fingerprint) & debug cases are taken care of?

junwang-noaa commented:
Yes, that would be good, this way we will keep the 2 debug test results.

On Tue, May 5, 2020 at 8:55 AM Minsuk Ji <notifications@github.com> wrote:

> For this PR, would it be safer if we generated the new baselines on Hera
> using nemscompsetrun? And start using rt.sh for baselines when the NEMS
> issue (verify fingerprint) & debug cases are taken care of?
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-s2s-model/pull/78#issuecomment-624038232>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TONWYLCFTCJTHNZZ2DRQAEEZANCNFSM4MXKY5RA>
> .
>

MinsukJi-NOAA committed:
*Add RT results from NEMSCompsetRun against new NEMSCompsetRun-generated baselines on Hera
*Remove REGTEST-FINGERPRINT.md from parm directory to disable verify_fingerprint

DeniseWorthen commented:
Is this PR ready to merge?

MinsukJi-NOAA commented:
> Is this PR ready to merge?
RT using rt.sh is pending

DeniseWorthen commented:
OK, great. I will wait for confirmation all tests you intend to run are complete.

MinsukJi-NOAA committed:
Add RT results from rt.sh

MinsukJi-NOAA commented:
@DeniseWorthen , all tests are now complete!
